### PR TITLE
Issue #90: Add ability to specify checksum flag in static compress methods

### PR DESCRIPTION
**Pull request by dineshjoshi**
@luben this patch adds the ability to specify a checksum on `Zstd::compress` and other static compress methods without the use of streams so they're GC friendly.

### CODE REVIEW

File: src/main/native/compress/zstd_compress.c [hunk 1]
```diff
@@ -3368,7 +3368,7 @@ size_t ZSTD_compress_usingDict(ZSTD_CCtx* cctx,
                          const void* dict, size_t dictSize,
                                int compressionLevel)
 {
-    ZSTD_parameters const params = ZSTD_getParams(compressionLevel, srcSize + (!srcSize), dict ? dictSize : 0);
+    ZSTD_parameters params = ZSTD_getParams(compressionLevel, srcSize + (!srcSize), dict ? dictSize : 0);
```
Comment (luben) [inline src/main/native/compress/zstd_compress.c]: Why this change here? I would prefer not to touch the upstream sources.

File: src/main/native/decompress/zstd_decompress.c [hunk 2]
```diff
@@ -614,6 +614,7 @@ static size_t ZSTD_decompressFrame(ZSTD_DCtx* dctx,
     }   }
     if (dctx->fParams.checksumFlag) { /* Frame content checksum verification */
         U32 const checkCalc = (U32)XXH64_digest(&dctx->xxhState);
+        DEBUGLOG(4, "ZSTD_decompressFrame computed checksum: %08X", checkCalc);
```
Comment (luben) [inline src/main/native/decompress/zstd_decompress.c]: Here also

**Comment by luben** [comment]
LGTM, plese remove the 2 lines that change upstream sources - I don't think we need them

**Comment by luben** [comment]
Thanks! Great contibution. I will release it soon on Maven

**Comment by dineshjoshi** [comment]
Oops looks like you merged before I squashed.

**Comment by dineshjoshi** [comment]
@luben how long before the jar is available on maven central?

**Comment by luben** [comment]
I just pushed it but ti may take a few hours before showing.

**Comment by luben** [comment]
https://repo1.maven.org/maven2/com/github/luben/zstd-jni/1.3.8-5/
